src/printer.js

```diff
@@ -3678,11 +3678,17 @@ function isBinaryish(node) {
 }
 
 function shouldInlineLogicalExpression(node) {
-  return (
-    node.type === "LogicalExpression" &&
-    (node.right.type === "ObjectExpression" ||
-      node.right.type === "ArrayExpression")
-  );
+  if (node.type === "LogicalExpression" &&
+      node.right.type === "ObjectExpression") {
+    return node.right.properties.length !== 0;
```

Comment:
Would be nice to put this complex expression inside of the if condition and return true. It's easier to understand.